### PR TITLE
Enable CI on Mac

### PR DESCRIPTION
**Pull request by Aurel300**
(draft for now to see if CI passes)

### CODE REVIEW

**Comment by Aurel300** [comment]
The failure seems to stem from the hard-coded `so` extension here:

https://github.com/viperproject/prusti-dev/blob/9a57e3d78d2539072d1b341efd34cfdb997fbadd/prusti/src/bin/prusti-rustc.rs#L68

File: prusti/src/bin/prusti-rustc.rs [hunk 1]
```diff
@@ -64,8 +64,13 @@ fn process(mut args: Vec<String>) -> Result<(), i32> {
         let prusti_contracts_path =
             get_latest_crate_artefact(&prusti_home, "prusti_contracts", "rlib");
         cmd.arg(format!("prusti_contracts={}", prusti_contracts_path));
+        let dylib_extension = if cfg!(target_os = "macos") {
+            "dylib"
```
Comment (vakaras) [inline prusti/src/bin/prusti-rustc.rs:68]: Isn't there a function in the standard library that returns the correct extension?

Comment (vakaras) [reply prusti/src/bin/prusti-rustc.rs:68]: Unfortunately, I could not find any.

**Review by vakaras — APPROVED** [review]
LGTM

**Pull request merged**